src/OmniSharp.Script/ScriptProjectSystem.cs

```diff
@@ -107,28 +131,43 @@ public void Initalize(IConfiguration configuration)
                     inheritedCompileLibraries.AddRange(DependencyContext.Default.CompileLibraries.Where(x =>
                             x.Name.ToLowerInvariant().StartsWith("system.runtime")));
                 }
-
             }
 
             // inject all inherited assemblies
             foreach (var inheritedCompileLib in inheritedCompileLibraries.SelectMany(x => x.ResolveReferencePaths()))
             {
-                Logger.LogDebug("Adding implicit reference: " + inheritedCompileLib);
-                AddMetadataReference(Context.CommonReferences, inheritedCompileLib);
+                _logger.LogDebug("Adding implicit reference: " + inheritedCompileLib);
+                AddMetadataReference(commonReferences, inheritedCompileLib);
             }
 
-            // Process each .CSX file
+            // Each .CSX file becomes an entry point for it's own project
+            // Every #loaded file will be part of the project too
             foreach (var csxPath in allCsxFiles)
             {
                 try
                 {
-                    CreateCsxProject(csxPath);
+                    var csxFileName = Path.GetFileName(csxPath);
+                    var project = ProjectInfo.Create(
+                        id: ProjectId.CreateNewId(Guid.NewGuid().ToString()),
```

Comment:
I don't think you need to create a new Guid for this. `ProjectId.Create()` already does that. The `debugName` parameter is to give it something human-readable for debugging. ([ref](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/ProjectId.cs,bac27363e2d822dc))